The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11096198459/job/30825749630) [(plain)](https://github.com/rust-lang/rust/commit/949a4daaa67a8d34ff04c7b83a47c9133d984d7c/checks/30825749630/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.974 Building wheels for collected packages: reuse
#13 2.975   Building wheel for reuse (pyproject.toml): started
#13 3.220   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.220   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132715 sha256=dfa09868353292d98f811d3efdb0d54d07389e808efc71d68e3b93c514bf8bec
#13 3.221   Stored in directory: /tmp/pip-ephem-wheel-cache-mptwuffq/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.223 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.626 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.627 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.158 Collecting virtualenv
#13 4.158 Collecting virtualenv
#13 4.224   Downloading virtualenv-20.26.6-py3-none-any.whl (6.0 MB)
#13 4.440      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.0/6.0 MB 28.1 MB/s eta 0:00:00
#13 4.501 Collecting filelock<4,>=3.12.2
#13 4.509   Downloading filelock-3.16.1-py3-none-any.whl (16 kB)
#13 4.545 Collecting platformdirs<5,>=3.9.1
#13 4.553   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.574 Collecting distlib<1,>=0.3.7
#13 4.591      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 57.5 MB/s eta 0:00:00
#13 4.591      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 57.5 MB/s eta 0:00:00
#13 4.671 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.862 Successfully installed distlib-0.3.8 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.26.6
#13 DONE 4.9s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      243648 kB
DirectMap2M:    11290624 kB
DirectMap1G:     7340032 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://static.rust-lang.org/dist/2024-09-22/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/2024-09-22/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt
---
##[endgroup]
fmt check
Diff in /checkout/library/core/src/slice/mod.rs:153:
     #[inline]
     #[must_use]
     pub const fn first(&self) -> Option<&T> {
-        if let [first, ..] = self {
-            Some(first)
-            None
-        }
-        }
+        if let [first, ..] = self { Some(first) } else { None }
 
     /// Returns a mutable reference to the first element of the slice, or `None` if it is empty.
Diff in /checkout/library/core/src/slice/mod.rs:180:
     #[inline]
     #[inline]
     #[must_use]
     pub const fn first_mut(&mut self) -> Option<&mut T> {
-        if let [first, ..] = self {
-            Some(first)
-            None
-        }
-        }
+        if let [first, ..] = self { Some(first) } else { None }
 
     /// Returns the first and all the rest of the elements of the slice, or `None` if it is empty.
Diff in /checkout/library/core/src/slice/mod.rs:204:
     #[inline]
     #[inline]
     #[must_use]
     pub const fn split_first(&self) -> Option<(&T, &[T])> {
-        if let [first, tail @ ..] = self {
-            Some((first, tail))
-            None
-        }
-        }
+        if let [first, tail @ ..] = self { Some((first, tail)) } else { None }
 
     /// Returns the first and all the rest of the elements of the slice, or `None` if it is empty.
Diff in /checkout/library/core/src/slice/mod.rs:230:
     #[inline]
     #[inline]
     #[must_use]
     pub const fn split_first_mut(&mut self) -> Option<(&mut T, &mut [T])> {
-        if let [first, tail @ ..] = self {
-            Some((first, tail))
-            None
-        }
-        }
+        if let [first, tail @ ..] = self { Some((first, tail)) } else { None }
 
     /// Returns the last and all the rest of the elements of the slice, or `None` if it is empty.
Diff in /checkout/library/core/src/slice/mod.rs:254:
     #[inline]
     #[inline]
     #[must_use]
     pub const fn split_last(&self) -> Option<(&T, &[T])> {
-        if let [init @ .., last] = self {
-            Some((last, init))
-            None
-        }
-        }
+        if let [init @ .., last] = self { Some((last, init)) } else { None }
 
     /// Returns the last and all the rest of the elements of the slice, or `None` if it is empty.
Diff in /checkout/library/core/src/slice/mod.rs:280:
     #[inline]
     #[inline]
     #[must_use]
     pub const fn split_last_mut(&mut self) -> Option<(&mut T, &mut [T])> {
-        if let [init @ .., last] = self {
-            Some((last, init))
-            None
-        }
-        }
+        if let [init @ .., last] = self { Some((last, init)) } else { None }
 
     /// Returns the last element of the slice, or `None` if it is empty.
Diff in /checkout/library/core/src/slice/mod.rs:303:
     #[inline]
     #[inline]
     #[must_use]
     pub const fn last(&self) -> Option<&T> {
-        if let [.., last] = self {
-            Some(last)
-            None
-        }
-        }
+        if let [.., last] = self { Some(last) } else { None }
 
     /// Returns a mutable reference to the last item in the slice, or `None` if it is empty.
Diff in /checkout/library/core/src/slice/mod.rs:330:
     #[inline]
     #[inline]
     #[must_use]
     pub const fn last_mut(&mut self) -> Option<&mut T> {
-        if let [.., last] = self {
-            Some(last)
-            None
-        }
-        }
+        if let [.., last] = self { Some(last) } else { None }
 
 
     /// Returns an array reference to the first `N` items in the slice.
Diff in /checkout/library/core/src/slice/mod.rs:3857:
         // Luckily since all this is constant-evaluated... performance here matters not!
         // Luckily since all this is constant-evaluated... performance here matters not!
         const fn gcd(a: usize, b: usize) -> usize {
-            if b == 0 {
-            } else {
-            } else {
-                gcd(b, a % b)
-            }
+            if b == 0 { a } else { gcd(b, a % b) }
 
         // Explicitly wrap the function call in a const block so it gets
Diff in /checkout/library/core/src/slice/mod.rs:4634:
 
 
         let offset = byte_offset / mem::size_of::<T>();
 
-        if offset < self.len() {
-            Some(offset)
-            None
-        }
-        }
+        if offset < self.len() { Some(offset) } else { None }
 
     /// Returns the range of indices that a subslice points to.
Diff in /checkout/library/core/src/slice/mod.rs:4692:
Diff in /checkout/library/core/src/slice/mod.rs:4692:
         let start = byte_start / core::mem::size_of::<T>();
         let end = start.wrapping_add(subslice.len());
 
-        if start <= self.len() && end <= self.len() {
-            Some(start..end)
-            None
-        }
-        }
+        if start <= self.len() && end <= self.len() { Some(start..end) } else { None }
 }
 
 
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/os/windows/mod.rs" "/checkout/library/std/src/os/windows/fs.rs" "/checkout/library/core/src/num/flt2dec/estimator.rs" "/checkout/library/core/src/num/wrapping.rs" "/checkout/library/core/src/num/f32.rs" "/checkout/library/core/src/num/int_log10.rs" "/checkout/library/core/src/ascii.rs" "/checkout/library/core/src/io/mod.rs" "/checkout/library/core/src/io/borrowed_buf.rs" "/checkout/library/std/src/os/windows/io/raw.rs" "/checkout/library/std/src/os/windows/io/mod.rs" "/checkout/library/std/src/os/windows/io/socket.rs" "/checkout/library/std/src/os/windows/io/tests.rs" "/checkout/library/std/src/os/windows/io/handle.rs" "/checkout/library/std/src/os/windows/thread.rs" "/checkout/library/std/src/os/windows/ffi.rs" "/checkout/library/std/src/os/windows/process.rs" "/checkout/library/core/src/convert/mod.rs" "/checkout/library/core/src/convert/num.rs" "/checkout/library/core/src/error.rs" "/checkout/library/std/src/os/redox/raw.rs" "/checkout/library/std/src/os/redox/mod.rs" "/checkout/library/std/src/os/redox/fs.rs" "/checkout/library/core/src/ptr/non_null.rs" "/checkout/library/core/src/ptr/mod.rs" "/checkout/library/core/src/ptr/unique.rs" "/checkout/library/core/src/ptr/metadata.rs" "/checkout/library/core/src/ptr/mut_ptr.rs" "/checkout/library/core/src/ptr/alignment.rs" "/checkout/library/core/src/ptr/const_ptr.rs" "/checkout/library/core/src/ascii/ascii_char.rs" "/checkout/library/std/src/os/wasi/net/mod.rs" "/checkout/library/std/src/os/wasi/mod.rs" "/checkout/library/std/src/os/wasi/fs.rs" "/checkout/library/std/src/os/wasi/io/raw.rs" "/checkout/library/std/src/os/wasi/io/mod.rs" "/checkout/library/std/src/os/wasi/io/fd.rs" "/checkout/library/std/src/os/wasi/io/fd/tests.rs" "/checkout/library/std/src/os/wasi/ffi.rs" "/checkout/library/core/src/slice/iter/macros.rs" "/checkout/library/core/src/slice/raw.rs" "/checkout/library/core/src/slice/mod.rs" "/checkout/library/core/src/slice/ascii.rs" "/checkout/library/core/src/slice/index.rs" "/checkout/library/core/src/slice/iter.rs" "/checkout/library/std/src/os/solid/mod.rs" "/checkout/library/std/src/os/solid/io.rs" "/checkout/library/std/src/os/solid/ffi.rs" "/checkout/library/std/src/os/uefi/mod.rs" "/checkout/library/std/src/os/uefi/env.rs" "/checkout/library/core/src/slice/sort/mod.rs" "/checkout/library/core/src/slice/sort/select.rs" "/checkout/library/std/src/os/vita/raw.rs" "/checkout/library/std/src/os/vita/mod.rs" "/checkout/library/std/src/os/vita/fs.rs" "/checkout/library/core/src/slice/sort/stable/mod.rs" "/checkout/library/core/src/slice/sort/stable/quicksort.rs" "/checkout/library/core/src/slice/sort/stable/tiny.rs" "/checkout/library/core/src/slice/sort/stable/merge.rs" "/checkout/library/core/src/slice/sort/stable/drift.rs" "/checkout/library/std/src/os/wasip2/mod.rs" "/checkout/library/core/src/slice/sort/shared/smallsort.rs" "/checkout/library/core/src/slice/sort/shared/mod.rs" "/checkout/library/std/src/os/windows/raw.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Sun Sep 29 21:59:06 UTC 2024
  network time: Sun, 29 Sep 2024 21:59:07 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/lib.rs
+++ b/library/core/src/lib.rs
@@ -146,7 +146,6 @@
 #![feature(const_size_of_val)]
 #![feature(const_size_of_val_raw)]
 #![feature(const_slice_from_ref)]
-#![feature(const_slice_split_at_mut)]
 #![feature(const_strict_overflow_ops)]
 #![feature(const_swap)]
 #![feature(const_try)]

--- a/library/core/src/slice/mod.rs
+++ b/library/core/src/slice/mod.rs
@@ -356,7 +356,8 @@ impl<T> [T] {
     /// ```
     #[inline]
     #[stable(feature = "slice_first_last_chunk", since = "1.77.0")]
-    #[rustc_const_unstable(feature = "const_slice_first_last_chunk", issue = "111774")]
+    #[rustc_const_stable(feature = "const_slice_first_last_chunk", since = "CURRENT_RUSTC_VERSION")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
     pub const fn first_chunk_mut<const N: usize>(&mut self) -> Option<&mut [T; N]> {
         if self.len() < N {
             None
@@ -421,7 +422,8 @@ impl<T> [T] {
     /// ```
     #[inline]
     #[stable(feature = "slice_first_last_chunk", since = "1.77.0")]
-    #[rustc_const_unstable(feature = "const_slice_first_last_chunk", issue = "111774")]
+    #[rustc_const_stable(feature = "const_slice_first_last_chunk", since = "CURRENT_RUSTC_VERSION")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
     pub const fn split_first_chunk_mut<const N: usize>(
         &mut self,
     ) -> Option<(&mut [T; N], &mut [T])> {
@@ -491,7 +493,8 @@ impl<T> [T] {
     /// ```
     #[inline]
     #[stable(feature = "slice_first_last_chunk", since = "1.77.0")]
-    #[rustc_const_unstable(feature = "const_slice_first_last_chunk", issue = "111774")]
+    #[rustc_const_stable(feature = "const_slice_first_last_chunk", since = "CURRENT_RUSTC_VERSION")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
     pub const fn split_last_chunk_mut<const N: usize>(
         &mut self,
     ) -> Option<(&mut [T], &mut [T; N])> {
@@ -560,7 +563,8 @@ impl<T> [T] {
     /// ```
     #[inline]
     #[stable(feature = "slice_first_last_chunk", since = "1.77.0")]
-    #[rustc_const_unstable(feature = "const_slice_first_last_chunk", issue = "111774")]
+    #[rustc_const_stable(feature = "const_slice_first_last_chunk", since = "CURRENT_RUSTC_VERSION")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
     pub const fn last_chunk_mut<const N: usize>(&mut self) -> Option<&mut [T; N]> {
         if self.len() < N {
             None
@@ -1903,7 +1907,8 @@ impl<T> [T] {
     #[inline]
     #[track_caller]
     #[must_use]
-    #[rustc_const_unstable(feature = "const_slice_split_at_mut", issue = "101804")]
+    #[rustc_const_stable(feature = "const_slice_split_at_mut", since = "CURRENT_RUSTC_VERSION")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
     pub const fn split_at_mut(&mut self, mid: usize) -> (&mut [T], &mut [T]) {
         match self.split_at_mut_checked(mid) {
             Some(pair) => pair,
@@ -2005,7 +2010,8 @@ impl<T> [T] {
     /// assert_eq!(v, [1, 2, 3, 4, 5, 6]);
     /// ```
     #[stable(feature = "slice_split_at_unchecked", since = "1.79.0")]
-    #[rustc_const_unstable(feature = "const_slice_split_at_mut", issue = "101804")]
+    #[rustc_const_stable(feature = "const_slice_split_at_mut", since = "CURRENT_RUSTC_VERSION")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
     #[inline]
     #[must_use]
     pub const unsafe fn split_at_mut_unchecked(&mut self, mid: usize) -> (&mut [T], &mut [T]) {
@@ -2105,7 +2111,8 @@ impl<T> [T] {
     /// assert_eq!(None, v.split_at_mut_checked(7));
     /// ```
     #[stable(feature = "split_at_checked", since = "1.80.0")]
-    #[rustc_const_unstable(feature = "const_slice_split_at_mut", issue = "101804")]
+    #[rustc_const_stable(feature = "const_slice_split_at_mut", since = "CURRENT_RUSTC_VERSION")]
+    #[cfg_attr(bootstrap, rustc_allow_const_fn_unstable(const_mut_refs))]
     #[inline]
     #[must_use]
     pub const fn split_at_mut_checked(&mut self, mid: usize) -> Option<(&mut [T], &mut [T])> {

